src/consumer.ts

```diff
@@ -91,69 +114,49 @@ export class Consumer {
     return this
   }
 
-  async on({ topic, fromBeginning }: ConsumerSubscribeTopic, callback: any) {
-    const callbackFn = this.resolveCallback(callback)
-    if (!callbackFn) {
-      throw new Error('no callback specified or cannot find your controller method')
+  async on(subscription: ConsumerSubscribeTopics, callback: ConsumerCallback): Promise<void>
+  async on(subscription: ConsumerSubscribeTopic, callback: ConsumerCallback): Promise<void>
+  async on(
+    subscription: ConsumerSubscribeTopic & ConsumerSubscribeTopics,
+    callback: ConsumerCallback
+  ) {
+    if (typeof callback !== 'function') {
+      throw new TypeError('Consumer callback is not a function')
     }
 
-    if (topic instanceof RegExp) {
-      throw new Error('regexp topic not supported by adonis-kafka yet')
+    let topics = []
+    if (Array.isArray(subscription.topics)) {
+      topics = subscription.topics
+    } else {
+      topics = subscription.topic.split(',').filter((topic) => !!topic)
     }
 
-    let topicArray = [topic]
-
-    if (typeof topic === 'string') {
-      topicArray = topic.split(',')
-    }
-
-    topicArray.forEach(async (item: any) => {
-      if (!item) {
-        return
-      }
-
+    topics.forEach(async (item) => {
       const events = this.events[item] || []
 
-      events.push(callbackFn)
+      events.push(callback)
 
       this.events[item] = events
 
       this.topics.push(item)
+    })
 
-      await this.consumer.subscribe({
-        topic: item,
-        fromBeginning: fromBeginning,
-      })
+    await this.consumer.subscribe({
+      topics,
+      fromBeginning: subscription.fromBeginning ?? false,
     })
   }
 
-  raiseError(topic: string, error: Error) {
+  handleError(topic: string, error: Error) {
     const handlers = this.errorHandlers[topic] || []
-    handlers.forEach((handler: any) => {
+    handlers.forEach((handler) => {
       handler(error)
     })
   }
```

Comment:
We may want to change `handleError` to `throw` if there are no handlers for the given topic, otherwise we risk swallowing errors.

Reply:
See #14